posting/views.py

```diff
@@ -1,22 +1,95 @@
+from posting.models import (
+        Posting,
+        PostingSize,
+        PostingHousing,
+        PostingStyle,
+        PostingSpace,
+        PostingLike,
+        PostingComment,
+        PostingScrap
+)
 
 class PostingView(View):
     def get(self, request):
-        postings        = Posting.objects.all()
+        postings = Posting.objects.all()
+        
+        sort_request = request.GET.get('sort', None)
+        postings = postings.annotate(like_num=Count("postinglike"))
+        postings = postings.annotate(comment_num=Count("comment"))
+        postings = postings.annotate(scrap_num=Count("postingscrap"))
+        sort_prefixes = {
+                "like"      : "-like_num",
+                "popular"   : "-comment_num",
+                "scrap"     : "-scrap_num",
+                "recent"    : "-created_at",
+                "old"       : "created_at"
+                }
+        if sort_request in sort_prefixes:
+            postings = postings.order_by(sort_prefixes[sort_request])
+
+        filter_prefixes = {
+                'size_id'       : 'size_id__in',
+                'housing_id'    : 'housing_id__in',
+                'style_id'      : 'style_id__in',
+                'space_id'      : 'space_id__in'
+                }
+        filter_set = {filter_prefixes.ge(key) : value for (key, value) in dict(request.GET).items() if filter_prefixes.get(key)}
+        postings = postings.filter(**filter_set)
+
         posting_list    = [{
-            "id"                        : posting.id,
-            "card_user_image"           : posting.user.imgage_url,
-            "card_user_name"            : posting.user.name,
-            "card_user_introduction"    : posting.user.description,
-            "card_image"                : posting.image_url,
-            "card_content"              : posting.content,
-            "comment_user_image"        : posting.postingcomment_set.first().user.image_url if posting.postingcomment_set.exists() else None,
-            "comment_user_name"         : posting.postingcomment_set.first().user.name if posting.postingcomment_set.exists() else None,
-            "comment_content"           : posting.postingcomment_set.first().content if posting.postingcomment_set.exists() else None,
-            } for posting in postings
+                "id"                        : posting.id,
+                "card_user_image"           : posting.user.image_url,
+                "card_user_name"            : posting.user.name,
+                "card_user_introduction"    : posting.user.description,
+                "card_image"                : posting.image_url,
+                "card_content"              : posting.content,
+                "comment_num"               : PostingComment.objects.filter(posting_id=posting.id).count() if posting.comment.exists() else 0,
+                "comment_user_image"        : posting.comment.first().user.image_url if posting.comment.exists() else None,
```

Comment:
```suggestion
                "card_content"              : posting.content,
                "comment_num"               : posting.comment.count() if posting.comment.exists() else 0,
                "comment_user_image"        : posting.comment.first().user.image_url if posting.comment.exists() else None,
```
다른 라인처럼 여기도 `related_name`바로 사용해보세요 :)

Reply:
넵!! related_name이 없어서 `_set`을 사용하겠습니다!!